File: Luogu/JCA/JC1/B_3713_语言月赛_202302_对了_还有花_少女_银河.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int n, m;
    cin >> n >> m;
    vector<string> tasks(m);
    for (auto &task : tasks) cin >> task;
    for (int i = 0; i < n; ++i) {
        string name, s, t;
        cin >> name;
        for (int j = 0; j < m; ++j) {
            cin >> s;
            t = name + ".zip/" + name + "/" + tasks[j] + "/" + tasks[j] + ".cpp";
            cout << (s == t ? "Fusu is happy!" : "Fusu is angry!") << endl;
        }
    }
    return 0;
}

File: Luogu/Course/JCA/JC1/B_3713_语言月赛_202302_对了_还有花_少女_银河.py
n, m = map(int, input().split())

tasks = [input() for _ in range(m)]

for _ in range(n):
    name = input()
    for task in tasks:
        s = input()
        t = f"{name}.zip/{name}/{task}/{task}.cpp"

        print("Fusu is happy!" if s == t else "Fusu is angry!")